devito/ops/transformer.py

```diff
@@ -169,13 +164,17 @@ def create_ops_dat(f, name_to_ops_dat, block):
             ops_dat_array,
             ListInitializer(dat_decls)
         )))
+
+        # Inserting the ops_dat array in case of TimeFunction.
+        name_to_ops_dat[f.name] = ops_dat_array
+
     else:
         ops_dat = OpsDat("%s_dat" % f.name)
         name_to_ops_dat[f.name] = ops_dat
 
-        d_p_val = tuple(sympify([p[0] + h[0] for p, h in zip(f.padding, f.halo)]))
-        d_m_val = tuple(sympify([-(p[1] + h[1]) for p, h in zip(f.padding, f.halo)]))
-        dim_shape = sympify(f.shape)
+        dim_shape = f.shape
```

Comment:
I now find it much more readable...don't you?

Reply:
Totally agree. When I was creating this feature it was real trouble understanding this part of the code so I could modify to my needs.